### PR TITLE
Add entity row that displays static text

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Add a new entity row to display static text.

Fixes https://github.com/home-assistant/core/issues/33846

```yaml
type: entities
title: My Title
entities:
  - type: text
    name: Hello
    text: World
    icon: mdi:party-popper
```

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by zsarnett** [comment]
I think text should be optional

**Comment by balloob** [comment]
It's… a text row? why would it be optional ?

File: src/panels/lovelace/special-rows/hui-text-row.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+} from "lit-element";
+
+import { LovelaceRow, TextConfig } from "../entity-rows/types";
+import { HomeAssistant } from "../../../types";
+
+import "../../../components/ha-icon";
+
+@customElement("hui-text-row")
+class HuiTextRow extends LitElement implements LovelaceRow {
+  public hass?: HomeAssistant;
```
Comment (zsarnett) [inline src/panels/lovelace/special-rows/hui-text-row.ts:18]: Do we need this here?

File: src/panels/lovelace/special-rows/hui-text-row.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  @property() private _config?: TextConfig;
+
+  public setConfig(config: TextConfig): void {
+    if (!config || !config.name || !config.text) {
+      throw new Error("Invalid Configuration: 'name' and 'url' required");
```
Comment (zsarnett) [inline src/panels/lovelace/special-rows/hui-text-row.ts:24]:
```suggestion
      throw new Error("Invalid Configuration: 'name' and 'text' required");
```

Comment (zsarnett) [reply src/panels/lovelace/special-rows/hui-text-row.ts:24]: Though I think text should be optional

File: src/panels/lovelace/create-element/create-row-element.ts [hunk 3]
```diff
@@ -41,6 +41,7 @@ const LAZY_LOAD_TYPES = {
   cast: () => import("../special-rows/hui-cast-row"),
   buttons: () => import("../special-rows/hui-buttons-row"),
   attribute: () => import("../special-rows/hui-attribute-row"),
+  text: () => import("../special-rows/hui-text-row"),
```
Comment (zsarnett) [inline src/panels/lovelace/create-element/create-row-element.ts:44]: Why do we need to Lazy load?

File: src/panels/lovelace/special-rows/hui-text-row.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config) {
+      return html``;
+    }
+
+    return html`
```
Comment (zsarnett) [inline src/panels/lovelace/special-rows/hui-text-row.ts:32]: Should we just use the generic row here to keep it the same as the others? Lest maintenance if it changes?

**Comment by zsarnett** [comment]
What if I wanted to just have text in the first column. As a title to the card or something. I see more use with not having the left value than having it. But just my POV

File: src/panels/lovelace/create-element/create-row-element.ts (hunk 3 above)
Comment (balloob) [reply src/panels/lovelace/create-element/create-row-element.ts:44]: Lazy load is the default. Only ones that are super popular should be considered to be bundled. Requires an editor too.

File: src/panels/lovelace/special-rows/hui-text-row.ts (hunk 4 above)
Comment (balloob) [reply src/panels/lovelace/special-rows/hui-text-row.ts:32]: generic row relies on an entity.

Comment (zsarnett) [reply src/panels/lovelace/special-rows/hui-text-row.ts:32]: Ah didnt realize

**Comment by balloob** [comment]
This row shouldn't be used for that. We have sections etc for that. If we allow that, people will want to add features to this row that will just blow up the scope. Let's keep it like this for now. You can always put a space for text ;)

**Comment by iantrich** [comment]
> What if I wanted to just have text in the first column. As a title to the card or something. I see more use with not having the left value than having it. But just my POV

And if someone really wanted that they can just do `text: ' '`

**Review by bramkragten — APPROVED** [review]

**Pull request merged**